solidity/test/integrator/TBTCDepositorProxy.test.ts

```diff
@@ -0,0 +1,317 @@
+  extraData:
+    "0xa9b38ea6435c8941d6eda6a46b68e3e2117196995bd154ab55196396b03d9bda",
+  expectedDepositKey:
+    "0xebff13c2304229ab4a97bfbfabeac82c9c0704e4aae2acf022252ac8dc1101d1",
+})
+
+describe("TBTCDepositorProxy", () => {
+  let bridge: MockBridge
+  let tbtcVault: MockTBTCVault
+  let depositorProxy: TestTBTCDepositorProxy
+
+  let fixture
+
+  before(async () => {
+    const MockBridge = await ethers.getContractFactory("MockBridge")
+    bridge = await MockBridge.deploy()
+
+    const MockTBTCVault = await ethers.getContractFactory("MockTBTCVault")
+    tbtcVault = await MockTBTCVault.deploy()
+
+    fixture = loadFixture(tbtcVault.address)
+
+    const TestTBTCDepositorProxy = await ethers.getContractFactory(
+      "TestTBTCDepositorProxy"
+    )
+    depositorProxy = await TestTBTCDepositorProxy.deploy()
+    await depositorProxy.initialize(bridge.address, tbtcVault.address)
+
+    // Assert that contract initializer works as expected.
+    await expect(
+      depositorProxy.initialize(bridge.address, tbtcVault.address)
+    ).to.be.revertedWith("TBTCDepositorProxy already initialized")
+  })
+
+  describe("initializeDeposit", () => {
+    context("when revealed vault does not match", () => {
+      it("should revert", async () => {
+        // Load the fixture with a different vault address.
+        const { fundingTx, reveal, extraData } = loadFixture(
+          ethers.constants.AddressZero
+        )
+
+        await expect(
+          depositorProxy.initializeDepositPublic(fundingTx, reveal, extraData)
+        ).to.be.revertedWith("Vault address mismatch")
+      })
+    })
+
+    context("when revealed vault matches", () => {
+      context("when deposit is rejected by the Bridge", () => {
+        before(async () => {
+          await createSnapshot()
+
+          // Pre-reveal the deposit to cause a revert on the second attempt
+          // made by the TBTCDepositorProxy.
+          await bridge.revealDepositWithExtraData(
+            fixture.fundingTx,
+            fixture.reveal,
+            fixture.extraData
+          )
+        })
+
+        after(async () => {
+          await restoreSnapshot()
+        })
+
+        it("should revert", async () => {
+          await expect(
+            depositorProxy.initializeDepositPublic(
+              fixture.fundingTx,
+              fixture.reveal,
+              fixture.extraData
+            )
+          ).to.be.revertedWith("Deposit already revealed")
+        })
+      })
+
+      context("when deposit is accepted by the Bridge", () => {
+        let tx: ContractTransaction
+
+        before(async () => {
+          await createSnapshot()
+
+          tx = await depositorProxy.initializeDepositPublic(
+            fixture.fundingTx,
+            fixture.reveal,
+            fixture.extraData
+          )
+        })
+
+        after(async () => {
+          await restoreSnapshot()
+        })
+
+        it("should reveal the deposit to the Bridge", async () => {
```

Comment:
A test to check that the deposit key is returned by `initializeDeposit` is missing.

Reply:
Addressed as part of a31abf46